File: src/JavaScript/Ones-and-Zeros.js
const sum = require("./toolkit/Statistics").sum;


const binaryArrayToNumber = arr => {
  // 向量化操作
  // return sum(arr.reverse().map((x, i) => x * Math.pow(2, i)));

  // 解析2进制字符串为10进制整数
  return parseInt(arr.join(''), 2);
};


console.log(binaryArrayToNumber([0, 1, 1, 0]) == 6);

File: src/Julia/Ones-and-Zeros.jl


using Test


"""
二进制各位数字向量转换为10进制数字
"""
function binaryarraytonumber(arr::Vector{Int64})::Int64
    # 解法一：二元函数 (i, x) 的向量化操作
    # [x * 2^(i - 1) for (i, x) ∈ enumerate(reverse(arr))] |> sum

    # 解法二：解析二进制字符串为10进制整数
    parse(Int, join(arr), base=2)
end


@test binaryarraytonumber([0, 1, 1, 0]) == 6